app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/AgendaDetailsScreen.kt

```diff
@@ -84,27 +85,26 @@ fun AgendaDetailsScreenRoot(
                     R.string.saved_successful,
                     Toast.LENGTH_LONG
                 ).show()
-                onBackClick()
+                onBackClick(selectedDateEpochDay)
             }
             AgendaDetailsEvent.DeleteSuccess -> {
                 Toast.makeText(
                     context,
                     R.string.delete_successful,
                     Toast.LENGTH_LONG
                 ).show()
-                onBackClick()
+                onBackClick(selectedDateEpochDay)
             }
         }
     }
     LaunchedEffect(Unit) {
         viewModel.updateStateWithEditTextArgs(editTextArgs)
     }
-    val state by viewModel.state.collectAsStateWithLifecycle()
     AgendaDetailsScreen(
         state = state,
         onAction = { action ->
             when (action) {
-                is AgendaDetailsAction.OnBackClick -> onBackClick()
+                is AgendaDetailsAction.OnBackClick -> onBackClick(null)
```

Comment:
It seems a bit weird for me to have to pass null here, do you guys think is there a better way to pass back the selectedDate?

Reply:
Why are you passing a date from the detail screen to the agenda list screen? 🤔 

AgendaList screen should keep the selected date as a state. When you return to the AgendaList screen, the state should still be there. You only need to pass the Agenda ID from the list screen to the detail screen. From there, you load the Agenda from the local storage via the ID.

Reply:
I'm thinking if the user change the date from the detail screen and create/update the item, to then display that date on the list.